### PR TITLE
Support overriding base name for metrics.

### PR DESCRIPTION
**Pull request by gshakhn**

### CODE REVIEW

File: src/main/java/io/vertx/ext/dropwizard/DropwizardMetricsOptions.java [hunk 1]
```diff
@@ -297,4 +298,12 @@ public DropwizardMetricsOptions addMonitoredHttpClientEndpoint(Match match) {
     return monitoredHttpClientEndpoints;
   }
 
+  public DropwizardMetricsOptions setBaseName(String baseName) {
+    this.baseName = baseName;
+    return this;
+  }
+
+  public String getBaseName() {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/dropwizard/DropwizardMetricsOptions.java]: javadoc

File: src/main/java/io/vertx/ext/dropwizard/impl/VertxMetricsFactoryImpl.java [hunk 2]
```diff
@@ -82,6 +83,14 @@ public VertxMetrics metrics(Vertx vertx, VertxOptions options) {
     return metrics;
   }
 
+  private String baseName(DropwizardMetricsOptions options) {
+    if (options.getBaseName() == null) {
```
Comment (vietj) [inline src/main/java/io/vertx/ext/dropwizard/impl/VertxMetricsFactoryImpl.java]: one liner ?

**Comment by vietj** [comment]
@gshakhn have you signed the Eclipse Contributor Agreement ?

**Comment by gshakhn** [comment]
@vietj I just signed the ECA.

Let me know if you would like other changes.

**Comment by tsegismont** [comment]
@gshakhn the changes look good to me. What about updating the package-info file so that the doc gets a new paragraph about that (with an example of setup and metric name change)?

**Comment by gshakhn** [comment]
@tsegismont: Added documentation about setup. Tried getting an example of a metric name change in the docs too, but the a few of the language converters couldn't convert it.

**Comment by tsegismont** [comment]
Not sure what you mean. It looks fine to me.

2016-11-14 14:34 GMT+01:00 George Shakhnazaryan <notifications@github.com>:

**Comment by gshakhn** [comment]
Wanted to document that (for example) `vertx.eventbus.handlers` became `foo.eventbus.handlers`. Hopefully understandable from context though.

**Comment by tsegismont** [comment]
I see. I reviewed the generated Asciidoc and actually it's fine with me.

So my +1

2016-11-14 14:45 GMT+01:00 George Shakhnazaryan <notifications@github.com>:

**Comment by vietj** [comment]
👍

**Pull request merged**